python/sources/InfluxDB-2.0/main.py

```diff
@@ -36,62 +56,38 @@ def interval_to_seconds(interval):
-# stream = producer_topic.create_stream("my-own-stream-id")  # To append data into the stream later, assign a stream id.
-stream_producer.properties.name = "influxdb-query"  # Give the stream a human readable name (for the data catalogue).
-stream_producer.properties.location = "/influxdb"  # Save stream in specific folder to organize your workspace.
-stream_producer.properties.metadata["version"] = "Version 1"  # Add stream metadata to add context to time series data.
-
-client = InfluxDBClient(token=os.environ["INFLUXDB_TOKEN"],
-                        url=os.environ["INFLUXDB_HOST"],
-                        org=os.environ["INFLUXDB_ORG"])
-query_api = client.query_api()
-
-measurement_name = os.environ.get("INFLUXDB_MEASUREMENT_NAME")
 interval = os.environ.get("task_interval", "5m")
 interval_seconds = interval_to_seconds(interval)
-
 bucket = os.environ["INFLUXDB_BUCKET"]
-org = os.environ["INFLUXDB_ORG"]
-
+measurement = os.environ['INFLUXDB_MEASUREMENT_NAME']
 
 # Function to fetch data from InfluxDB and send it to Quix
 # It runs in a continuous loop, periodically fetching data based on the interval.
 def get_data():
-    # Run in a loop until the main thread is terminated
+    # Run in a loop until the main thread is terminated |> range(start: -{interval})
     while True:
         start = datetime.now().timestamp()
         flux_query = f'''
             from(bucket: "{bucket}")
                 |> range(start: -{interval})
+                |> filter(fn: (r) => r._measurement == "{measurement}")
                 |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
             '''
-        df = query_api.query_data_frame(org=org, query=flux_query)
+        influx_df = query_api.query_data_frame(query=flux_query,org=os.environ['INFLUXDB_ORG'])
+
+        # Convert Timestamp columns to ISO 8601 formatted strings to avoid serialization errors
+        datetime_cols = influx_df.select_dtypes(include=['datetime64[ns]', 'datetime64[ns, UTC]']).columns
+        for col in datetime_cols:
+            influx_df[col] = influx_df[col].dt.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
 
         # If the query returns tables with different schemas, the result will be a list of dataframes.
-        if isinstance(df, list):
-            for item in df:
-                item["timestamp"] = pd.to_datetime(item["_time"])
-                stream_producer.timeseries.buffer.publish(item)
+        if isinstance(influx_df, list):
+            for item in influx_df:
+                yield item
                 print(f"Published 1 row to Quix")
-        elif isinstance(df, pd.DataFrame) and not df.empty:
-            df["timestamp"] = pd.to_datetime(df["_time"])
-            stream_producer.timeseries.buffer.publish(df)
-            print(f"Published {len(df)} rows to Quix")
+        elif isinstance(influx_df, pd.DataFrame) and not influx_df.empty:
+            yield influx_df
+            print(f"Published {len(influx_df)} rows to Quix")
```

Comment:
Very minor suggestion: Might wanna be careful with "row" terminology, especially if people are used to the old QuixStreams...they might get the impression that multiple rows are contained in the same message.

I think if we could just change "rows" to "messages", or maybe even "Published N rows as messages to Quix"